modules/apps/application-list/application-list-api/src/main/java/com/liferay/application/list/PanelAppRegistry.java

```diff
@@ -111,13 +111,6 @@ public List<PanelApp> getPanelApps(String parentPanelCategoryKey) {
 			panelApp -> {
 				Portlet portlet = panelApp.getPortlet();
 
-				if (portlet == null) {
-					portlet = _portletLocalService.getPortletById(
-						panelApp.getPortletId());
-
-					panelApp.setPortlet(portlet);
-				}
-
 				if (portlet == null) {
 					return false;
 				}
```

Comment:
@W2jx @dantewang After we changed the structure of PanelApp, passed in portlet or referenced portlet, which mean once you have an instance of PanelApp, the _portlet attribute should not be null. So I think if we do null checking in the constructor should be enough, and here, we should consider it is never null.

Reply:
I will fix it.